cognite/neat/rules/exporter/rules2graphql.py

```diff
@@ -8,17 +9,36 @@
     from typing_extensions import Self
 
 from cognite.neat.rules import exceptions
-from cognite.neat.rules._validation import (
+from cognite.neat.rules.analysis import to_class_property_pairs
+from cognite.neat.rules.exporter._validation import (
     are_entity_names_dms_compliant,
     are_properties_redefined,
 )
-from cognite.neat.rules.analysis import to_class_property_pairs
-from cognite.neat.rules.models import DATA_TYPE_MAPPING, TransformationRules
+from cognite.neat.rules.models.rules import DATA_TYPE_MAPPING, Rules
 from cognite.neat.utils.utils import generate_exception_report
 
+from ._base import BaseExporter
+
 if TYPE_CHECKING:
     from jinja2 import Template
 
+
+class GraphQLSchemaExporter(BaseExporter):
+    def __init__(self, rules: Rules, filepath: Path):
+        self.rules = rules
+        self.filepath = filepath
+        self.data = GraphQLSchema.from_rules(self.rules).schema
+
+        super().__init__(self.rules, self.filepath)
+
+    def export(self, filepath: Path | None = None):
+        filepath = filepath or self.filepath
+        if not filepath:
+            raise ValueError("No filepath given")
+
+        filepath.write_text(self.data)
```

Comment:
I think the init only should set up the configuration, the actual logic for conversion should be in the export function. 
```suggestion

        super().__init__(self.rules, self.filepath)

    def export(self, filepath: Path | None = None):
        filepath = filepath or self.filepath
        if not filepath:
            raise ValueError("No filepath given")
        data = GraphQLSchema.from_rules(self.rules).schema
        filepath.write_text(data)
```

Reply:
done